src/Services/Gateway/HpsCreditService.php

```diff
@@ -395,6 +396,9 @@ public function listTransactions($startDate, $endDate, $filterBy = null)
         $hpsReportActivity = $xml->createElement('hps:ReportActivity');
         $hpsReportActivity->appendChild($xml->createElement('hps:RptStartUtcDT', $startDate));
         $hpsReportActivity->appendChild($xml->createElement('hps:RptEndUtcDT', $endDate));
+        if ($deviceId != null) {
```

Comment:
Let's use strict inequality here (`!==`) or check for non-empty (`!empty($deviceId)`). The choice between those two should depend on how the gateway treats empty values for the `DeviceId` data element.